### ISSUE TITLE
Problem calculating average true range

### ISSUE THREAD

**Issue by luisen14** [bukosabino/ta#25]
Hola,

First thank you for making this library available to the rest of us, second I have problem using the average_true_range function. As an example, I just tried to calculate the same values of the example you put as a reference:

https://stockcharts.com/school/doku.php?id=chart_school:technical_indicators:average_true_range_atr

But I couldn't get the same result as the refered page. I made sure my values were right, and finally I think the problem lies in the utils.ema function, and the way it calls pandas.ewm function. 

I got it to produce the desired values, by calling the ewm function and specify the decay in terms of center of mass, like this:

return  pd.concat([sma, rest]).ewm(com=(self.periods-1), adjust=False).mean() 

instead of calling it like:

return pd.concat([sma, rest]).ewm(span=periods, adjust=False).mean()

why do you have the calculation of the average using the decay in terms of span?

**Comment by bukosabino** [comment]
Hi @luisen14 ,

You are right. I cannot reproduce the results of this link: https://stockcharts.com/school/doku.php?id=chart_school:technical_indicators:average_true_range_atr

I test this issue in this notebook: https://github.com/bukosabino/ta/blob/develop/dev/average_true_range_test.ipynb
I get the same results for TR, but we calculate EMA in a different way. I would like to study it deeply.

BTW, we are discussing how to implement EMA on this issue: https://github.com/bukosabino/ta/issues/22

**Comment by bukosabino** [comment]
Hi @luisen14 ,


I can reproduce the 'stockcharts school' results with this new implementation:


```python
def average_true_range(high, low, close, n=14, fillna=False):
    """Average True Range (ATR)

    The indicator provide an indication of the degree of price volatility.
    Strong moves, in either direction, are often accompanied by large ranges,
    or large True Ranges.

    http://stockcharts.com/school/doku.php?id=chart_school:technical_indicators:average_true_range_atr

    Args:
        high(pandas.Series): dataset 'High' column.
        low(pandas.Series): dataset 'Low' column.
        close(pandas.Series): dataset 'Close' column.
        n(int): n period.

    Returns:
        pandas.Series: New feature generated.
    """
    cs = close.shift(1)
    tr = high.combine(cs, max) - low.combine(cs, min)

    atr = np.zeros(len(close))
    atr[0] = tr[1::].mean()
    for i in range(1, len(atr)):
        atr[i] = (atr[i-1] * 13 + tr[i]) / 14

    if fillna:
        atr = atr.replace([np.inf, -np.inf], np.nan).fillna(0)

    return pd.Series(atr, name='atr')

```

**Comment by bukosabino** [comment]
This code is implemented in the last ta library version (0.3.4).

Regards.

**Comment by adityamertia** [comment]
I still don't get accurate results for ATR using this library. 
The code that gets me accurate ATR is as below:
df = Date, High, Low, Close
```
def avg_true_range(df, minPeroids, Window): 
  ind = range(0,len(df))
  indexlist = list(ind)
  df.index = indexlist

  for index, row in df.iterrows():
    if index != 0:
      tr1 = row["High"] - row["Low"]
      tr2 = abs(row["High"] - df.iloc[index-1]["Close"])
      tr3 = abs(row["Low"] - df.iloc[index-1]["Close"])
      true_range = max(tr1, tr2, tr3)
      df.at[index,"TR"] = true_range

  return df["TR"].rolling(min_periods=minPeroids, window=Window, center=False).mean()
```
